### ISSUE TITLE
Implement v1 of ADAS in AutoGen

### ISSUE THREAD

**Issue by yeandy** [microsoft/autogen#4569]
Feature to dynamically generate new agents using AutoGen code. This is based on the ADAS paper [here](https://arxiv.org/pdf/2408.08435).

Original discussion here https://github.com/microsoft/autogen/discussions/4486.

**Comment by ekzhu** [comment]
@yeandy would you like to create a project and label and use it to track your progress?

**Comment by yeandy** [comment]
Yes, that would be great! I don't have permission to create; would you be able to assist? something like a `adas` label should be fine for now I think.

**Comment by ekzhu** [comment]
@yeandy  I just created a label for this. For project I think you can create one.

**Comment by yeandy** [comment]
Thanks!
